File: AspNetCoreApiStarter.Model/Reservation.cs
using System;

namespace AspNetCoreApiStarter.Model
{
    public class Reservation
    {
        public int Id { get; set; }

        public string Username { get; set; }

        public int RoomId { get; set; }

        public DateTime Date { get; set; }

        public int Slot { get; set; }
    }
}

File: AspNetCoreApiStarter/Site/src/app/features/reservations/models/reservation.ts
export class Reservation {
    id: number;
    username: string;
    roomId: number;
    date: string;
    slot: number;

    constructor() {
        this.id = 0;
        this.username = '';
        this.roomId = 0;
        this.date = '';
        this.slot = 0;
    }
}
